clang/include/clang/Basic/Builtins.td

```diff
@@ -1244,6 +1244,12 @@ def ElementwiseATan : Builtin {
   let Prototype = "void(...)";
 }
 
+def ElementwiseATan2 : Builtin {
+  let Spellings = ["__builtin_elementwise_atan2"];
+  let Attributes = [NoThrow, Const, CustomTypeChecking];
```

Comment:
This is more a note for me:  We haven't been doing `CustomTypeChecking` for `"__builtin_hlsl_elementwise_*"` cases. Check to see if we should. I'm assuming `CustomTypeChecking` is used to have vector cases avoid impacting variadic argument cases.